File: 41-first-missing-positive/41-first-missing-positive.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        unordered_map<int,int>mp;
        for(auto i:nums) mp[i]++;
        
        int k=1;
        while (mp[k]) k+=1;
        
        return k;
    }
};

File: 41-first-missing-positive/41-first-missing-positive.py
from collections import defaultdict as dfd
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        d = dfd(int)
        for i in nums:
            d[i] =1
        for i in range(1,2**31):
            if(d[i] == 0):
                return i